duckduckgo/Android
# Insert DefaultBrowser Cta and SearchWidget Cta as Dax dialog during Dax Journey

Task/Issue URL: https://app.asana.com/0/488551667048375/1161278038072567/f
Tech Design URL: 
CC: 

**Description**:
In this experiment we are going to place our CTAs inside our Dax Journey. Our CTAs will match the style/tone of the existing Dax dialogs.

- Default Browser CTA: After the user visits a site with trackers, we prompt a Dax dialog to tell the user that we blocked all the trackers in that site. After showing that dialog we will ask the user to set us as Default browser.

- Search Widget CTA: When the user visits DuckDuckGo, a Dax dialog will appear informing our users that all their searches are private. After showing that dialog we will ask the user to add our search widget to protect their future searches.

**Concept test variant mj: Vanila concept test**

1. Hardcode into variantManager variant **"mj"**
1. Perform a fresh install
1. Run the app and ensure onboarding doesn't show any cta
1. Visit a web with trackers like cnn.com
1. Ensure dax dialog appears informing about trackers found
1. Click on "High five!"
1. Ensure no more Ctas are shown
1. Visit duckduckgo.com
1. Ensure dax dialog appears
1. Click on "Phew!"
1. Ensure no more Ctas are shown

**Concept test variant ml: Concept test with old ctas**

1. Hardcode into variantManager variant **"ml"**
1. Perform a fresh install
1. Run the app and ensure during the onboarding Default browser cta appears
1. Visit a web with trackers like cnn.com
1. Ensure dax dialog appears informing about trackers found
1. Click on "High five!"
1. Ensure no more Ctas are shown
1. Visit duckduckgo.com
1. Ensure dax dialog appears
1. Click on "Phew!"
1. Ensure no more Ctas are shown
1. Continue the Dax journey till the end (open new tab, visit a website, open a new tab)
1. Ensure Search widget cta appears in home screen

**Concept test variant mh: concept test with ctas as dax dialogs**:

> Dax Default Browser cta:

- Test-1

1. Hardcode into variantManager variant **"mh"**
2. Perform a fresh install
3. Run the app and ensure onboarding doesn't show any default browser cta
4. Visit a web with trackers like cnn.com
5. Ensure dax dialog appears informing about trackers found
6. Click on "High five!"
7. Ensure DefaultBrowser cta appears 
8. Ensure pixel "m_odc_s" is fired with "dbd" as param
9. Click on "Ok"
10. Ensure pixel "m_odc_ok" is fired with "dbd" as param
11. Ensure Dialog asking to set ddg as default browser appears

_Repeat each of the following scenarios with previous steps:_

a) 
12. Click on "just once"
13. Ensure dialog appears again
14. Ensure no pixel is fired
14. Click on "always"
15. Ensure dialog dissapears
16. Ensure pixel "m_db_s" is fired with "dbo=d" as param 

b) 
12. Click on "just once"
13. Ensure dialog appears again
14. Click on "always"
15. Ensure dialog dissapears
16. Ensure pixel "m_db_ns" is fired with "dbo=jom" as param 

c) 
12. press back
13. Ensure dialog is dismissed
14. Ensure pixel "m_db_ns" is fired with "dbo=dd" as param 

d) 
12. select another browser
13. Go back to DDG app
14. Ensure dialog is dismissed
14. Ensure pixel "m_db_ns" is fired with "dbo=e" as param 

- Test-2

1. Hardcode into variantManager variant **"mh"**
2. Perform a fresh install
3. Go to settings and set as another browser as default
4. Run the app and ensure onboarding doesn't show any default browser cta
5. Visit a web with trackers like cnn.com
6. Ensure dax dialog appears informing about trackers found
7. Click on "High five!"
8. Ensure DefaultBrowser cta appears 
9. Ensure pixel "m_odc_s" is fired with "dbs" as param


_Repeat each of the following scenarios with previous steps:_

a)
10. Click on "Go to settings"
11. Ensure pixel "m_odc_ok" is fired with "dbs" as param
12. Ensure user navigates to settings
13. Select ddg as default browser
14. Go back to the app
15. Ensure pixel "m_db_s" is fired with "fo=false, dbo=dxs" as param
16. Ensure cta is gone

b)
10. Click on "Maybe later"
11. Ensure pixel "m_odc_c" is fired with "cta=dbs" as param
12. Ensure cta is gone


> Dax Search widget cta:

- Test-1

1. Hardcode into variantManager variant **"mh"**
1. Perform a fresh install
1. Run the app and ensure onboarding doesn't show any default browser cta
1. Visit duckduckgo.com
1. Ensure dax dialog appears
1. Click on "Phew!"
1. Ensure SearchWidget cta appears 
1. Ensure pixel "m_odc_s" is fired with "wa" as param
1. Click on "Add"
1. Ensure pixel "m_odc_ok" is fired with "wa" as param
1. Ensure cta is gone
1. Ensure dialog appears asking the user to add the search widget

- Test-2

1. Hardcode into variantManager variant **"mh"**
1. Perform a fresh install
1. Run the app and ensure onboarding doesn't show any default browser cta
1. Visit duckduckgo.com
1. Ensure dax dialog appears
1. Click on "Phew!"
1. Ensure SearchWidget cta appears 
1. Ensure pixel "m_odc_s" is fired with "wa" as param
1. Click on "Maybe later"
1. Ensure pixel "m_odc_c" is fired with "wa" as param
1. Ensure cta is gone

- Test-3 (device running api < 23)

1. Hardcode into variantManager variant **"mh"**
1. Perform a fresh install
1. Run the app and ensure onboarding doesn't show any default browser cta
1. Visit duckduckgo.com
1. Ensure dax dialog appears
1. Click on "Phew!"
1. Ensure no cta appears

- Test-4 (device running api 23/24/26)

1. Hardcode into variantManager variant **"mh"**
1. Perform a fresh install
1. Run the app and ensure onboarding doesn't show any default browser cta
1. Visit duckduckgo.com
1. Ensure dax dialog appears
1. Click on "Phew!"
1. Ensure SearchWidget cta appears 
1. Ensure pixel "m_odc_s" is fired with "wm" as param
1. Click on "Add"
1. Ensure pixel "m_odc_ok" is fired with "wm" as param
1. Ensure cta is gone
1. Ensure dialog appears showing the user how to add the search widget

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
